tensorflow_datasets/core/features/rle_feature/rle/tf_impl.py

```diff
@@ -0,0 +1,101 @@
+"""Tensorflow encode/decode/utility implementations."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+major, minor = (int(v) for v in tf.__version__.split('.')[:2])
+is_over_1_12 = major >= 1 and minor > 12
+
+
+def ensure_supported_version(error_identifier):
```

Comment:
Let's move this logic to `tf_compat.py` and add a function that can be called as `tf_compat.assert_tensorflow_version_gt("1.12.0", err_msg)`.